data/dlist/basic.lean

```diff
@@ -0,0 +1,6 @@
+
+import data.dlist
+
+def dlist.join {α : Type*} : list (dlist α) → dlist α
+ | [] := dlist.empty
+ | (x :: xs) := x ++ dlist.join xs
```

Comment:
This file needs a header. I don't think it needs its own folder; it can go in `data/dlist.lean` except that conflicts with core, so how about `data/dlist2.lean`? That's what I did for `data/vector2.lean`.

Reply:
I agree with @cipher1024. A directory makes sense here, and it's a better convention than create files with numerical suffixes.

Reply:
I'm fine with this.